File: api/src/main/java/net/impactdev/gts/api/events/auctions/BidEvent.java
package net.impactdev.gts.api.events.auctions;

import net.impactdev.impactor.api.event.ImpactorEvent;
import net.impactdev.gts.api.listings.Listing;
import net.impactdev.impactor.api.event.type.Cancellable;
import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;

import java.util.UUID;

@GenerateFactoryMethod
public interface BidEvent extends ImpactorEvent, Cancellable {

    UUID getBidder();

    Listing getListing();

    double getAmountBid();

}

File: api/src/main/kotlin/net/impactdev/gts/api/events/auctions/BidEvent.kt
package net.impactdev.gts.api.events.auctions

import net.impactdev.gts.api.listings.Listing
import net.impactdev.impactor.api.event.ImpactorEvent
import net.impactdev.impactor.api.event.type.Cancellable
import java.util.*

interface BidEvent : ImpactorEvent, Cancellable {
    @get:Param(0)
    val bidder: UUID

    @get:Param(1)
    val listing: Listing

    @get:Param(2)
    val amountBid: Double
}